share/wake/lib/system/path.wake

```diff
@@ -17,10 +17,10 @@
 global def read path =
   def imp p = prim "read"
   match (getPathResult path)
-    Pass f = match (imp f)
+    Pass f = match (imp f.getPathName)
       Pass body = Pass body
       Fail f = Fail (makeError f)
-    x = x
+    Fail e = Fail e
```

Comment:
This one was fun, I figured it out quickly but adding the `.getPathName` had no effect on the reported type error which was a bit surprising!